lookout/core/tests/test_lib.py

```diff
@@ -80,23 +81,30 @@ def test_files_by_language(self):
         self.assertEqual({"js": 2, "python": 5, "ruby": 7}, {k: len(v) for k, v in result.items()})
         return result
 
-    def test_filter_files(self):
-        files = {"one.py": File(content=b"hello"), "two.py": File(content=b"world" * 100)}
-        logged = False
+    def test_parse_files(self):
 
         class Log:
             def debug(self, *args, **kwargs):
                 nonlocal logged
                 logged = True
 
-        filtered = filter_files(files, 80, 1000000, log=Log())
-        self.assertEqual(len(filtered), 1)
-        self.assertEqual(filtered[0].content, b"hello")
-        self.assertTrue(logged)
-        filtered = filter_files(files, 80, 1, log=Log())
-        self.assertEqual(len(filtered), 0)
-        filtered = filter_files(files, 80, 5 * 100, log=Log())
-        self.assertEqual(len(filtered), 1)
+        logged = False
+        with NamedTemporaryFile(prefix="one", suffix=".js") as tmp1:
+            tmp1.write(b"hello")
+            tmp1.seek(0)
+            with NamedTemporaryFile(prefix="two", suffix=".js") as tmp2:
+                tmp2.write(b"world" * 100)
+                tmp2.seek(0)
+            try:
+                bblfsh_client = BblfshClient("0.0.0.0:9432")
+                filtered = parse_files(filepaths=[tmp1.name, tmp2.name], line_length_limit=80,
+                                       overall_size_limit=5 << 20, client=bblfsh_client,
+                                       language="javascript", log=Log())
+                self.assertEqual(len(filtered), 1)
+                self.assertEqual(filtered[0].content, b"hello")
+                self.assertTrue(logged)
+            finally:
+                bblfsh_client._channel.close()
```

Comment:
Can you please add a test for filtering by max line length? You can take samples from the test files in `dirname(__file__)`

Reply:
Isn't that a bad practice in general as those test files can be modified later ?

Reply:
Yeah, not a rock solid solution, but I think the way you coded it is fine